### PR TITLE
rec: implement rfc6303 special zones (mostly v6 reverse mappings)

### PR DESCRIPTION
**Pull request by omoerbeek**
### Short description

Guarded by a `recursor.serve_rfc6303` setting, default `true`.

Requested by @neilcook 

### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [X] tested this code
- [X] included documentation (including possible behaviour changes)
- [X] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

File: pdns/recursordist/settings/table.py [hunk 1]
```diff
@@ -2361,6 +2361,18 @@
...
+        'name' : 'serve_rfc6303',
+        'section' : 'recursor',
+        'type' : LType.Bool,
+        'default' : 'true',
+        'help' : 'If we should be authoritative for RFC 6303 private IP space',
+        'doc' : '''
+This makes the server authoritatively aware of the zones in RFC 6303 not covered by RFC 1918.
+Individual parts of these zones can still be loaded or forwarded.
+''',
+        'versionadded': ['5.1.x', '5.2.0'],
```
Comment (rgacogne) [inline pdns/recursordist/settings/table.py:2374]:
```suggestion
        'versionadded': ['5.1.3', '5.2.0'],
```

File: pdns/recursordist/reczones.cc [hunk 2]
```diff
@@ -508,6 +511,37 @@ static void processServeRFC1918(std::shared_ptr<SyncRes::domainmap_t>& newMap, L
...
+static void processServeRFC6303(std::shared_ptr<SyncRes::domainmap_t>& newMap, Logr::log_t log)
+{
+  if (!::arg().mustDo("serve-rfc6303")) {
+    return;
+  }
+  SLOG(g_log << Logger::Warning << "Inserting rfc 6303 private space zones" << endl,
+       log->info(Logr::Notice, "Inserting rfc 6303 private space zones"));
+  // Section 4.2
+  makePartialIPZone(*newMap, {"0"}, log);
+  // makePartialIPZone(*newMap, { "127" }, log) already done in processServeRFC1918
```
Comment (rgacogne) [inline pdns/recursordist/reczones.cc:523]: Are we okay with what happens if `serve-rfc6303` is enabled but `serve-rfc1918` is not?

File: pdns/recursordist/reczones.cc [hunk 3]
```diff
@@ -199,6 +200,7 @@ string reloadZoneConfiguration(bool yaml)
         ::arg().preParseFile(filename, "allow-notify-for-file", ::arg()["allow-notify-for-file"]);
         ::arg().preParseFile(filename, "export-etc-hosts", ::arg()["export-etc-hosts"]);
         ::arg().preParseFile(filename, "serve-rfc1918", ::arg()["serve-rfc1918"]);
+        ::arg().preParseFile(filename, "serve-rfc1918", ::arg()["serve-rfc6303"]);
```
Comment (rgacogne) [inline pdns/recursordist/reczones.cc:203]:
```suggestion
        ::arg().preParseFile(filename, "serve-rfc6303", ::arg()["serve-rfc6303"]);
```

**Review by rgacogne — CHANGES_REQUESTED** [review]

**Review by rgacogne — APPROVED** [review]

File: pdns/recursordist/reczones.cc (hunk 2 above)
Comment (omoerbeek) [reply pdns/recursordist/reczones.cc:523]: I'll look into that. Easiest thing would be to forbid that.

**Comment by mnordhoff** [comment]
Updates #8679. :D

**Pull request merged**